### ISSUE TITLE
[import/order] Breaking change behavior in v2.27

### ISSUE THREAD

**Issue by StefanoMagrassi** [import-js/eslint-plugin-import#2722]
In version 2.27.0 it was introduced a [new strategy](https://github.com/import-js/eslint-plugin-import/compare/v2.26.0...v2.27.0#diff-2825fcc427c846fe18d501930e1a3d83be722027f1f643801afdcc2c255dd4a8R285) which splits the imports paths in segments and compares them independently.

However, this leads to different results for the same inputs in some cases, for example:

```ts
import {screen} from '@testing-library/react';
import {checkDropdown} from '@testing/form';
```
this is the "right" order as by v2.26 because:
```ts
compareString('@testing-library/react', '@testing/form') === -1
```
but in v2.27 it is reported as error because:
```ts
compareString('@testing-library', '@testing') === 1
```
I don't know if the current behavior is the right one, but it is definitely a breaking change

**Comment by ljharb** [comment]
I agree that it was an unintentional change, and also that the one that’s right in 2.26 is the right one. (i think the change you’re referencing was #2396)

**Comment by StefanoMagrassi** [comment]
> (i think the change you’re referencing was #2396)

yep

**Comment by 7iomka** [comment]
Any news on this?

**Comment by ljharb** [comment]
No, not yet, but it'd be great to get a PR up to fix it.

**Comment by 7iomka** [comment]
> No, not yet, but it'd be great to get a PR up to fix it.

Just revert #2396

**Comment by ljharb** [comment]
@7iomka it may not be that simple; there may be some fixes in it that we'd need to keep.

**Comment by DaveTryon** [comment]
This is still broken in 2.28. We're hitting it in https://github.com/microsoft/accessibility-insights-web/pull/6846. `git bisect` confirms that it broke in #2396 

Here's a quick unit test to use as a reference:

```
    // Option alphabetize: {order: 'asc'} when folders and paths start with the same substring
    test({
      code: `
        import a from "foo/foobar-file";
        import b from "foo/foobar/file";
      `,
      options: [{ alphabetize: { order: 'asc' } }],
    }),
```

**Comment by tw3** [comment]
PR: https://github.com/import-js/eslint-plugin-import/pull/3196
